contracts/extensions/OneTxPayment.sol

```diff
@@ -60,108 +51,74 @@ contract OneTxPayment is ColonyExtension {
   /// @param _childSkillIndex Index of the _permissionDomainId skill.children array to get
   /// @param _callerPermissionDomainId The domainId in which the _caller_ has permissions to add a payment and fund it
   /// @param _callerChildSkillIndex Index of the _callerPermissionDomainId skill.children array to get
   /// @param _workers The addresses of the recipients of the payment
   /// @param _tokens Addresses of the tokens the payments are being made in. 0x00 for Ether.
   /// @param _amounts amounts of the tokens being paid out
-  /// @param _domainId The Id of the domain the payment should be coming from
-  /// @param _skillId The Id of the skill that the payment should be marked with, possibly awarding reputation in this skill.
+  /// @param _domainId The domainId the payment should be coming from
+  /// @param _skillId The skillId that the payment should be marked with, possibly awarding reputation in this skill.
   function makePayment(
-    uint256 _permissionDomainId,
+    uint256 _permissionDomainId, // Unused
     uint256 _childSkillIndex,
-    uint256 _callerPermissionDomainId,
-    uint256 _callerChildSkillIndex,
+    uint256 _callerPermissionDomainId, // Unused
+    uint256 _callerChildSkillIndex, // Unused
     address payable[] memory _workers,
     address[] memory _tokens,
     uint256[] memory _amounts,
     uint256 _domainId,
-    uint256 _skillId) public
+    uint256 _skillId
+  )
+    public
   {
-    // Arrays must be of equal size
-    require(
-      _workers.length == _tokens.length && _workers.length == _amounts.length,
-      "colony-one-tx-payment-arrays-must-be-equal-length"
-    );
-    // Check caller is able to call {add,finalize}Payment and moveFundsBetweenPots on the colony
-    validateCallerPermissions(_callerPermissionDomainId, _callerChildSkillIndex, _domainId);
-    // In addition, check the caller is able to call moveFundsBetweenPots from the root domain
-    require(
-      ColonyAuthority(colony.authority()).canCall(msg.sender, 1, address(colony), MOVE_FUNDS_SIG),
-      "colony-one-tx-payment-root-funding-not-authorized"
-    );
-    // Get around stack too deep with array
-    // paymentData[0] = expenditure slot
-    // paymentData[1] = expenditureId/paymentId
-    // paymentData[3] = fundingPotId
-    uint256[3] memory paymentData;
-    paymentData[0] = 0;
-    if (_tokens.length > 1) {
-      // Make a new expenditure
-      paymentData[1] = colony.makeExpenditure(_permissionDomainId, _childSkillIndex, _domainId);
-      ColonyDataTypes.Expenditure memory expenditure = colony.getExpenditure(paymentData[1]);
-      paymentData[2] = expenditure.fundingPotId;
-      colony.setExpenditureRecipient(paymentData[1], paymentData[0], _workers[0]);
-      colony.setExpenditureSkill(paymentData[1], paymentData[0], _skillId);
+    require(_workers.length == _tokens.length && _workers.length == _amounts.length, "colony-one-tx-payment-invalid-input");
+    validatePermissions(1, _childSkillIndex, _domainId);
```

Comment:
I suppose I was trying to simplify... an alternative would be to adapt `_callerPermissionDomainId` to specifically concern the administration permission (since funding must exist in root).